lineapy/graph_reader/program_slice.py

```diff
@@ -24,6 +30,14 @@ def get_slice_graph(
     :return: A subgraph extracted (i.e., sliced) for the desired node IDs.
 
     """
+    for node in graph.nodes:
+        # If there is a control flow node which has a branch which was never encountered during runtime,
+        # we take a conservative approach and include all of the lines present in the unexecuted code block
+        # so that in case the branch is visited during production, we do not alter the user's intended behavior
+        if isinstance(node, ControlFlowNode):
+            if node.unexec_id is not None:
+                sinks.append(node.id)
```

Comment:
This is on purpose, to ensure in case the unexec block had a statement which affected the artifact, all possible operations are included to ensure the artifact in question's value is not affected in the sliced version.